### DISCUSSION TITLE
AWSIM Installation and Setup

### DISCUSSION THREAD

**Discussion by npnquang** [Monash-Connected-Autonomous-Vehicle/autoware#24] [Show and tell]
@Monash-Connected-Autonomous-Vehicle/avp-team 
Hi, I'm bumping my own message on the software channel.

I have set up the full version of AWSIM running with Unity since we have been using the demo. I followed this [tutorial](https://tier4.github.io/AWSIM/GettingStarted/SetupUnityProject/) and the AWSIM is working fine. I want to make some notes:
- The AWSIM has its own ROS version (ROS2forUnity) so we should not let it run while the local ROS2 is sourced. Hence, I commented out the source command in the `.bashrc` file, which means we must manually source it whenever we open a new terminal.
- When the AWSIM is running, I ran ros2 topic list command and we do not have any ROS2 topics running, so now the Autoware cannot communicate with the AWSIM anymore. I need some help to fix this, so if anyone has any ideas please tell me.
![image](https://github.com/Monash-Connected-Autonomous-Vehicle/autoware/assets/73055557/99ab95b4-a3e9-41ca-bed6-f1845a8c53e4)

Thanks

**Comment by npnquang** [comment]
Hi, I found the solution to this problem so I'll close this discussion.
1. We need to run the following command to run the AWSIM with Unity Hub: `~/Unity/Hub/Editor/2021.1.7f1/Editor/Unity -projectPath .`
2. Open another terminal, source the local version of ROS, source the autoware and run it normally.
`source /opt/ros/humble/setup.bash
cd autoware`
`source install/setup.bash`
`ros2 launch autoware_launch e2e_simulator.launch.xml vehicle_model:=sample_vehicle sensor_model:=awsim_sensor_kit map_path:=/home/mcav/awsim_map`

**Reply by dylan-gonzalez** [reply]
Why are we running AWSIM from within Unity again?

Also sorry for not responding earlier, but glad you got it figured out

**Reply by npnquang** [reply]
The Demo only gives us limited functionalities, so I think we should set it up with Unity. One more reason is we still need to fix the "empty" RecognitionResultOnImage section, so I think we should have the full version of AWSIM to debug it.

**Reply by dylan-gonzalez** [reply]
yep true you're right.
Also, I've updated the wiki with instructions for running Autoware [here](https://github.com/Monash-Connected-Autonomous-Vehicle/autoware/wiki/Autoware)

Can you double check it and make any changes if needed? Thanks!

**Reply by npnquang** [reply]
The instructions is all good. I have just added a small note about the `projectPath` flag of the command to run the AWSIM with Unity. One more thing, I don't think we need to run this command `rocker --nvidia --x11 --user --volume $HOME/autoware --volume $HOME/autoware_map -- ghcr.io/autowarefoundation/autoware-universe:latest-cuda`, as we do not to launch the Docker container to run the Autoware. Do you have any specific reason to put it there?

**Reply by dylan-gonzalez** [reply]
So we installed Autoware on the Beauty and we were getting an error with rmw_implementation when trying to launch Autoware.

We realised that if you want to do a _source_ installation, you have to install the [RMW Implementation](https://github.com/autowarefoundation/autoware/tree/main/ansible/roles/rmw_implementation#manual-installation). But otherwise, the docker container should already have these things installed.

So my theory as to why we can only run Autoware from within the docker container on the Beauty, and why we can it run it inside and outside the docker container on the Beast is because someone might have started following the instructions to install from source on the Beast and installed all the prerequisite tools outlined on the [source installation steps](https://autowarefoundation.github.io/autoware-documentation/main/installation/autoware/source-installation/).

**Reply by npnquang** [reply]
Ok understood. Should we figure it out how to run Autoware and AWSIM inside a Docker container?

**Reply by dylan-gonzalez** [reply]
Eventually but not a priority right now.

**Comment by dylan-gonzalez** [comment]
@npnquang 

On the Beast, AWSIM from within Unity cannot talk properly to Autoware from inside the docker container - was this the issue we were having?

@jxcksonli @KehanRana to get Autoware working with AWSIM on the Beast, don't go inside the docker container - skip this step. (bit puzzling as to why it was working inside the docker container before but now doesn't)

**Reply by dylan-gonzalez** [reply]
pinging rest of the team @Monash-Connected-Autonomous-Vehicle/avp-team in case you need to run Autoware + AWSIM on the Beast

**Reply by npnquang** [reply]
I didn't bump into that issue the last time. Let me Anydesk to the Beast and check it.

**Reply by npnquang** [reply]
.Ah I see what you mean. Yes, I haven't tried to launch a Docker container with Unity. Just run the autoware locally on our machine and it will work. We need to change the instructions [here](https://github.com/Monash-Connected-Autonomous-Vehicle/autoware/wiki/Autoware) as well.

**Reply by npnquang** [reply]
I have edited the above instruction to make it more clear. Sorry for the confusion.
